### DISCUSSION TITLE
Use OkHttpHttpSender to send a list of objects in JSON format.

### DISCUSSION THREAD

**Discussion by chukunx** [open-telemetry/opentelemetry-java#6700] [Q&A] [answered]
Hi OTel friends, it looks like when the `OkHttpHttpSender` is configured to [`exportAsJson`](https://github.com/open-telemetry/opentelemetry-java/blob/v1.42.0/exporters/sender/okhttp/src/main/java/io/opentelemetry/exporter/sender/okhttp/internal/OkHttpHttpSender.java#L211), the Marshaler [delegate JsonSerializer](https://github.com/open-telemetry/opentelemetry-java/blob/v1.42.0/exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/Marshaler.java#L30) to serialize the object, and it always enclose the message with [`{`](https://github.com/open-telemetry/opentelemetry-java/blob/v1.42.0/exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/JsonSerializer.java#L221) and [`}`](https://github.com/open-telemetry/opentelemetry-java/blob/v1.42.0/exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/JsonSerializer.java#L223). Do we support serializing lists, i.e. enclosing the body with `[` and `]`?

**Comment by chukunx** [comment]
Followup: Would you open to remove the `final` keyword of the [`writeJsonTo`](https://github.com/open-telemetry/opentelemetry-java/blob/v1.42.0/exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/Marshaler.java#L28-L39) methods so that this behavior is customizable when using OkHttpHttpSender to send JSON objects?

**Reply by jack-berg** [reply, marked as answer]
`exportAsJson` is not part of the public API, so we don't document or make guarantees about its behavior. 

Everywhere we do JSON serialization of OTLP payloads, we follow the [OTLP spec](https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#json-protobuf-encoding).

**Reply by chukunx** [reply]
I see, the sender deals with OTLP request (and response) so it doesn't make much sense to make it customizable at this level. Thanks!
